apps/site/lib/site_web/controllers/route_controller.ex

```diff
@@ -3,10 +3,46 @@ defmodule SiteWeb.RouteController do
   Endpoints for getting route data.
   """
   use SiteWeb, :controller
-  alias Routes.Repo
+  alias RoutePatterns.RoutePattern
+  alias Routes.{Repo, Route}
 
   def get_by_stop_id(conn, %{"stop_id" => stop_id} = _params) do
-    routes = Repo.by_stop(stop_id)
-    json(conn, routes)
+    routesWithPolylines =
+      stop_id
+      |> Repo.by_stop()
+      |> Enum.map(fn route ->
+        route
+        |> Route.to_json_safe()
+        |> Map.put(:polylines, route_polylines(route, stop_id))
+      end)
+
+    json(conn, routesWithPolylines)
+  end
+
+  defp route_polylines(route, stop_id) do
+    if is_rail_route?(route) or Route.silver_line?(route) do
+      route.id
+      |> RoutePatterns.Repo.by_route_id(stop: stop_id)
+      |> Enum.map(fn %RoutePattern{shape_id: id, representative_trip_polyline: polyline} ->
```

Comment:
suggestion: this could be a good spot to deduplicate the shape IDs:
```suggestion
      |> Enum.uniq_by(& &1.shape_id)
      |> Enum.map(fn %RoutePattern{shape_id: id, representative_trip_polyline: polyline} ->
```

Reply:
Ah, wrong place for it - there aren't duplicate shapes within a single route (which is the context for this particular function, but it comes into play when serving several routes together).